lua/telescope/make_entry.lua

```diff
@@ -899,28 +899,48 @@ function make_entry.gen_from_autocommands(_)
   end
 end
 
-function make_entry.gen_from_git_status(_)
+function make_entry.gen_from_git_status(opts)
+  opts = opts or {}
+
   local displayer = entry_display.create {
-  separator = " ",
+  separator = " ", -- NOTE: this is an &nbsp which fixes alignment whilst icons are not being rendered correctly
   items = {
-      { width = 1},
-      { width = 1},
+      { width = 2},
+      { width = 2},
       { remaining = true },
     }
   }
 
+  local giticons = opts.git_icons or {}
   local make_display = function(entry)
-    local modified = "TelescopeResultsDiffChange"
-    local staged = "TelescopeResultsDiffAdd"
+    local icons = {
+      add       = giticons.added     or "+",
+      change    = giticons.changed   or "~",
+      delete    = giticons.deleted   or "-",
+      untracked = giticons.untracked or "?"
+    }
+
+    local str_1 = string.sub(entry.status, 1, 1)
+    local str_2 = string.sub(entry.status, -1)
+
+    local git_abbrev = {
+      ["A"] = icons.add,
+      ["M"] = icons.change,
+      ["D"] = icons.change,
```

Comment:
Since we can distinguish between modified and deleted now, we could use the deleted icon instead?

Reply:
ah that's a copy/paste oversight. fixed